themes/default/content/learn/pulumi-fundamentals/configure-and-provision/index.md

```diff
@@ -0,0 +1,463 @@
+                                      )]
+                                      )
+```
+
+{{% /choosable %}}
+
+With Docker networking, we can use image names to refer to a container. In our
+example, the React frontend client sends requests to the Express backend client.
+The URL to the backend is set via the `setupProxy.js` file in the
+`app/frontend/src` directory with the `HTTP_PROXY` environment variable.
+
+Run `pulumi up` to get the application running. However, the store is empty, and
+we need to add products to the database.
+
+## Populate the database
+
+Now we can populate MongoDB and set up our Pulumi program to autopopulate the
+next time we deploy. First, copy the `products.json` file into the same
+directory as your {{% langfile %}} file.
+
+```bash
+cp app/data/products.json .
+```
+
+Then, we'll mount the file to an ephemeral seed container, and then use
+`mongoimport` to transfer that data into the database. Add the following lines
+of code to your Pulumi file, then run `pulumi up`.
+
+Add this snippet after the `backend_container` declaration:
+
+{{< chooser language "python" / >}}
+
+{{% choosable language python %}}
+
+```python
+data_seed_container = docker.Container("data_seed_container",
+                                       image=mongo_image.latest,
+                                       name="data_seed",
+                                       must_run=False,
+                                       rm=True,
+                                       opts=pulumi.ResourceOptions(depends_on=[backend_container]),
+                                       mounts=[docker.ContainerMountArgs(
+                                           target="/home/products.json",
+                                           type="bind",
+                                           source=f"{os.getcwd()}/products.json"
+                                       )],
+                                       command=[
+                                           "sh", "-c",
+                                           "mongoimport --host mongo --db cart --collection products --type json --file /home/products.json --jsonArray"
+                                       ],
+                                       networks_advanced=[docker.ContainerNetworksAdvancedArgs(
+                                           name=network.name
+                                       )]
+                                       )
+```
+
+{{% /choosable %}}
+
+Note the `mounts` part, which allows you to use a `bind mount` storage type to
+add the necessary file. If you're not familiar with mounts compared to volumes
+in Docker, see [the docs on bind
+mounts](https://docs.docker.com/storage/bind-mounts/). In this case, we're using
+a bind mount over a volume for simplicity.
+
+Once you've added this snippet, run `pulumi up` to refresh the data in the
+database.
+
+Open a browser to `http://localhost:3001`, and our application is now deployed.
```

Comment:
:/ Hm. I'm going to revisit this after launch; I've got a remote image that has the data pre-imported that we'll probably switch to in order to avoid this kind of error in the future. We built that image for the cloud fundamentals pathways, but I may as well use it here for a better experience.

Reply:
https://github.com/pulumi/pulumi-hugo/issues/837